docs/machine-learning/tutorials/image-classification-api-transfer-learning.md

```diff
@@ -0,0 +1,451 @@
+    ```
+
+1. Then, load the images into an [`IDataView`](xref:Microsoft.ML.IDataView) using the [`LoadFromEnumerable`](xref:Microsoft.ML.DataOperationsCatalog.LoadFromEnumerable*) method,
+
+    ```csharp
+    IDataView imageData = mlContext.Data.LoadFromEnumerable(images);
+    ```
+
+1. The way in which the data is loaded is in the order it was read from the directories. To balance the data, shuffle it using the [`ShuffleRows`](xref:Microsoft.ML.DataOperationsCatalog.ShuffleRows*) method.
+
+    ```csharp
+    IDataView shuffledData = mlContext.Data.ShuffleRows(imageData);
+    ```
+
+1. Machine learning models expect input to be in numerical format. Therefore, some pre-processing needs to be done on the data prior to training. Create an [`EstimatorChain`](xref:Microsoft.ML.Data.EstimatorChain%601) made up of the [`MapValueToKey`](xref:Microsoft.ML.ConversionsExtensionsCatalog.MapValueToKey*) and [`LoadImages`](xref:Microsoft.ML.ImageEstimatorsCatalog.LoadImages*) transforms. The `MapValueToKey` transform takes the categorical value in the `Label` column, converts it to a numerical `KeyType` value and stores it in a new column called `LabelAsKey`. The `LoadImages` takes the values from the `ImagePath` column along with the `imageFolder` parameter to load images for training. Setting the `useImageType` to `false` converts the images into a `byte[]`. 
+
+    ```csharp
+    var preprocessingPipeline = mlContext.Transforms.Conversion.MapValueToKey(
+            inputColumnName:"Label",
+            outputColumnName:"LabelAsKey")
+        .Append(mlContext.Transforms.LoadImages(
+            outputColumnName:"Image", 
+            imageFolder: assetsRelativePath,
+            useImageType: false,
+            inputColumnName:"ImagePath"));
+    ```
+
+1. Use the [`Fit`](xref:Microsoft.ML.Data.EstimatorChain%601.Fit*) method to apply the data to the `preprocessingPipeline` [`EstimatorChain`](xref:Microsoft.ML.Data.EstimatorChain%601) followed by the [`Transform`](xref:Microsoft.ML.Data.TransformerChain`1.Transform*) method which returns an [`IDataView`](xref:Microsoft.ML.IDataView) containing the pre-processed data.
+
+    ```csharp
+    IDataView preProcessedData = preprocessingPipeline
+                                    .Fit(shuffledData)
+                                    .Transform(shuffledData);
+    ```
+
+1. To train a model, it's important to have a training dataset as well as a validation dataset. The model is trained on the training set of data and how well it makes predictions on unseen data is measured by the performance against the validation set. Based on the results of that performance, the model makes adustments to what it has learned in an effort to improve. The validation set can come from either splitting your original dataset or from another source that has already been set aside for this purpose. In this case, the pre-processed dataset is split into training, validation and test sets.
```

Comment:
Suggestion:

To train a model, it's important to have a training dataset as well as a validation dataset. The model is trained on the training set. How well it makes predictions on unseen data is measured by the performance against the validation set.

Reply:
#resolved